conwayste/src/ui/layer.rs

```diff
@@ -31,137 +32,245 @@ use super::{
 
 use crate::constants::colors::*;
 
-pub struct Layer {
-    pub id: WidgetID,
-    pub widgets: Vec<Box<dyn Widget>>,
+pub struct Layering {
     pub with_transparency: bool,
-    pub focused_widget: Option<WidgetID>,
+    pub widget_list: Vec<BoxedWidget>,
+    focused_ids: Vec<Option<WidgetID>>,
+    id_cache: Vec<WidgetID>
 }
 
-/// A container of one or more widgets or panes
-impl Layer {
-    /// Specify the unique widget identifer for the layer
-    pub fn new(widget_id: WidgetID) -> Self {
-        Layer {
-            id: widget_id,
-            widgets: vec![],
+/// A Layering is a container of one or more widgets or panes, where widgets are ordered by
+/// by their z_index to create the appearance of a layer, for a given game screen. Layerings
+/// support an optional transparency between two adjacent z-orders.
```

Comment:
Is there one `Layering` per screen? Might want to say whether or not that's the case.